py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,739 @@
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunrise'],
+        })
+
+    def _format_sunset(self, wthr):
+        # Get the time for sunset (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNSET, 0))
+
+        # Format the sunset
+        replaced = dt.strftime(self.format_sunset)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunset'],
+        })
+
+    def _format_dict(self, wthr):
+        return {
+            # Standard options
+            'icon': self._get_icon(wthr),
+            'clouds': self._format_clouds(wthr),
+            'rain': self._format_rain(wthr),
+            'snow': self._format_snow(wthr),
+            'wind': self._format_wind(wthr),
+            'humidity': self._format_humidity(wthr),
+            'pressure': self._format_pressure(wthr),
+            'temperature': self._format_temp(wthr),
+            'sunrise': self._format_sunrise(wthr),
+            'sunset': self._format_sunset(wthr),
+
+            # Descriptions (defaults to empty)
+            'desc': self._jpath(wthr, OWM_DESC, '').lower(),
+            'desc_long': self._jpath(wthr, OWM_DESC_LONG, '')
+        }
+
+    def _format(self, wthr, fcsts):
+        # Format all sections
+        today = self._format_dict(wthr)
+
+        # Insert forecasts
+        forecasts = []
+        for day in fcsts:
+            future = self._format_dict(day)
+            forecasts.append(self.py3.safe_format(self.format_forecast, future))
+
+        # Give the final format
+        today['forecast'] = self.py3.composite_join(
+            self.forecast_text_separator, forecasts)
+
+        return self.py3.safe_format(self.format, today)
+
+    def weather_owm(self):
+        # Get weather information
+        coords = self._get_coords()
+        text = ''
+        if coords is not None:
+            wthr = self._get_weather(coords)
+            fcsts = self._get_forecast(coords)
+
+            text = self._format(wthr, fcsts)
+
+        return {
+            'full_text': text,
+            'cached_until': self.py3.time_in(seconds=self.cache_timeout)
+        }
+
+
+if __name__ == '__main__':
+    '''
+    Run module in test mode.
+    '''
+
+    import os
+    from py3status.module_test import module_test
+
+    # All possible outputs
+    all_string = '/'.join([
+        '{clouds}',
+        '{desc_long}',
+        '{desc}',
+        '{humidity}',
+        '{pressure}',
+        '{snow}',
+        '{sunrise}',
+        '{sunset}',
+        '{temperature}',
+        '{wind}'
+    ])
+
+    module_test(Py3status, config={
+        'api_key': os.getenv('OWM_API_KEY'),
+
+        # Select icons
+        'icons': {
+            'i200': "☔",
+            'i230_i232': "🌧",
+
+            'clouds': "☁",
+        },
+
+        # Complete configuration
+        'format_cloud': '{icon}: {coverage}%',
```

Comment:
Super happy to have `{location}` placeholder. It worked great. Things changed when I played with `(lat, long)`. Still uneasiness because we don't have `{city.name}` placeholder. Would be ideal for laptops on the go... and possibly good to have it part of the default format too. This module is getting bigger and bigger. :o

And this is a typo. We might want to rename the others to `format_cloud` to prevent little things like this. I also see `clouds` for `icons`. Maybe it should be `cloud` too.